userspace/libsinsp/sinsp_filtercheck_fspath.cpp

```diff
@@ -374,20 +369,79 @@ uint8_t* sinsp_filter_check_fspath::extract_single(sinsp_evt* evt,
 		case PPME_SYSCALL_LINKAT_X:
 			enter_param = evt->get_enter_evt_param("oldpath");
 			if(!enter_param.has_value()) {
-				return NULL;
+				return false;
 			}
 			m_tstr = enter_param.value();
 			break;
 		default:
 			if(!extract_fspath(evt, extract_values, m_target_checks)) {
-				return NULL;
+				return false;
 			}
 			m_tstr.assign((const char*)extract_values[0].ptr,
 			              strnlen((const char*)extract_values[0].ptr, extract_values[0].len));
 		};
 		break;
 	default:
-		return NULL;
+		return false;
+	}
+	return true;
+}
+
+bool sinsp_filter_check_fspath::extract_new_implementation(sinsp_evt* evt) {
+	switch(m_field_id) {
+	case TYPE_NAME:
+	case TYPE_NAMERAW:
+		switch(evt->get_type()) {
+		case PPME_SYSCALL_OPEN:
+			ASSERT(strncmp(evt->get_param_name(1), "name", 5) == 0);
+			m_tstr = evt->get_param(1)->as<std::string>();
+			break;
+
+		default:
+			return false;
+		};
+		break;
+	case TYPE_SOURCE:
+	case TYPE_SOURCERAW:
+		switch(evt->get_type()) {
+		default:
+			return false;
+		};
+		break;
+	case TYPE_TARGET:
+	case TYPE_TARGETRAW:
+		switch(evt->get_type()) {
+		default:
+			return false;
+		};
+		break;
+	default:
+		return false;
+	}
+	return true;
+}
+
+uint8_t* sinsp_filter_check_fspath::extract_single(sinsp_evt* evt,
+                                                   uint32_t* len,
+                                                   bool sanitize_strings) {
+	*len = 0;
+	ASSERT(evt);
+
+	// todo!: cleanup this workaround when the support will be complete
+	if(evt->is_new_event_version()) {
+		switch(evt->get_type()) {
+		case PPME_SYSCALL_OPEN:
+			if(!extract_new_implementation(evt)) {
```

Comment:
No need to switch this, right? The switch is already inside `extract_new_implementation`.

Reply:
done, ty!